apiserver/facades/agent/secretsmanager/secrets.go

```diff
@@ -430,11 +436,57 @@ func (s *SecretsManagerAPI) GetSecretContentInfo(args params.GetSecretContentArg
 			contentParams.Data = content.SecretValue.EncodedValues()
 		}
 		result.Results[i].Content = contentParams
-		result.Results[i].BackendConfig = backend
+		if backend != nil {
+			result.Results[i].BackendConfig = &params.SecretBackendConfigResult{
+				ControllerUUID: backend.ControllerUUID,
+				ModelUUID:      backend.ModelUUID,
+				ModelName:      backend.ModelName,
+				Draining:       draining,
+				Config: params.SecretBackendConfig{
+					BackendType: backend.BackendType,
+					Params:      backend.Config,
+				},
+			}
+		}
 	}
 	return result, nil
 }
 
+func (s *SecretsManagerAPI) getRemoteSecretContent(uri *coresecrets.URI, refresh, peek bool, label string, updateLabel bool) (
+	*secrets.ContentParams, *secretsprovider.ModelBackendConfig, bool, error,
+) {
+	extClient, err := s.remoteClientGetter(uri)
+	if err != nil {
+		return nil, nil, false, errors.Annotate(err, "creating remote secret client")
+	}
+	consumerApp := authTagApp(s.authTag)
+	token, err := s.crossModelState.GetToken(names.NewApplicationTag(consumerApp))
+	if err != nil {
+		return nil, nil, false, errors.Annotatef(err, "getting remote token for %q", consumerApp)
+	}
+	var untiId int
+	if unitTag, ok := s.authTag.(names.UnitTag); ok {
+		untiId = unitTag.Number()
+	} else {
+		return nil, nil, false, errors.NotSupportedf("getting cross model secret for consumer %q", s.authTag)
+	}
+	content, backend, draining, err := extClient.GetRemoteSecretContentInfo(uri, refresh, peek, token, untiId)
+	if err != nil {
+		return nil, nil, false, errors.Trace(err)
+	}
+	if updateLabel {
+		consumerInfo, err := s.secretsConsumer.GetSecretConsumer(uri, s.authTag)
+		if err != nil {
+			return nil, nil, false, errors.Trace(err)
+		}
+		consumerInfo.Label = label
+		if err := s.secretsConsumer.SaveSecretConsumer(uri, s.authTag, consumerInfo); err != nil {
```

Comment:
So consumer data is stored on the consumer controller side.

Reply:
The apiserver logic uses the SourceUUID in the secret URI to figure out what model to look at. So we need to resolve the mapping from the label to the URI in the consuming model.
I'll double check the logic to make sure it's correct.

Reply:
There was a logic error - the cross model facade was saving consumer data and it should not have been, so fixed that.